pkg/clean/types.go

```diff
@@ -3,3 +3,27 @@ package clean
 type cleanup interface {
 	Cleanup() error
 }
+
+type CleanupService struct {
+	cleans []cleanup
+	err    error
+}
+
+func NewCleanupService() *CleanupService {
+	return &CleanupService{}
+}
+
+func (c *CleanupService) AddCleanupService(f func() (cleanup, error)) *CleanupService {
+	var clean cleanup
```

Comment:
seems no need to define a var here, just use := in line 22?

Reply:
<img width="708" alt="image" src="https://github.com/vmware-tanzu/nsx-operator/assets/4385934/3981ac78-f323-410a-ad3b-a2cd6a741bec">
no , since I use  `c.err = f()` not `err:=f()`